src/operator/publishReplay.ts

```diff
@@ -6,13 +6,15 @@ import { publishReplay as higherOrder } from '../operators/publishReplay';
 /**
  * @param bufferSize
  * @param windowTime
+ * @param selector
  * @param scheduler
- * @return {ConnectableObservable<T>}
+ * @return {Observable<T> | ConnectableObservable<T>}
  * @method publishReplay
  * @owner Observable
  */
 export function publishReplay<T>(this: Observable<T>, bufferSize: number = Number.POSITIVE_INFINITY,
```

Comment:
Like below, we'll need to add some function overloads for TypeScript users to keep this change from being breaking. Here's what the `publish` counterpart to this file is doing...

https://github.com/ReactiveX/rxjs/blob/accbcd0c5f9fd5976be3f491d454c4a61f699c4b/src/operator/publish.ts#L7-L9